libr/core/cmd_anal.c

```diff
@@ -4244,7 +4244,7 @@ static void cmd_anal_syscall(RCore *core, const char *input) {
 	case 'c': // "asc"
 		if (input[1] == 'a') {
 			if (input[2] == ' ') {
-				if ((n = atoi (input + 2)) > 0) {
+				if ((n = strtol (input + 2, &out, 10 )) >= 0 && *out == '\0') {
```

Comment:
Better use rnumcalc to support resolving flags and math ops. Just heck if first char is not a alpha